### ISSUE TITLE
Paddle Paddle训练求解

### ISSUE THREAD

**Issue by zhu143xin** [PaddlePaddle/PaddleFormers#535]
strategy = hub.AdamWeightDecayStrategy(
        warmup_proportion=args.warmup_proportion,
        weight_decay=args.weight_decay,
        learning_rate=args.learning_rate)

    # Setup runing config for PaddleHub Finetune API
    config = hub.RunConfig(
            eval_interval=args.eval_step,
            save_ckpt_interval=args.model_save_step,
            use_data_parallel=args.use_data_parallel,
            use_cuda=args.use_gpu,
            num_epoch=args.num_epoch,
            batch_size=args.batch_size,
            checkpoint_dir=args.checkpoint_dir,
            strategy=strategy)

    # Define a sequence labeling finetune task by PaddleHub's API
    # If add crf, the network use crf as decoder
    seq_label_task = hub.SequenceLabelTask(
        data_reader=reader,
        feature=sequence_output,
        feed_list=feed_list,
        max_seq_len=args.max_seq_len,
        num_classes=dataset.num_labels,
        config=config,
        add_crf=args.add_crf)
以上是使用BERT+CRF进行序列标注任务，后续调用finetune_and_eval进行训练。想问一下能否自己单独设置CRF的学习率，让它与BERT不相同，目前没找到实现的方法

**Comment by kinghuin** [comment]
Hi，您可以参考 https://github.com/PaddlePaddle/PaddleHub/blob/release/v1.6/paddlehub/finetune/strategy.py#L389
通过判断param.name的方式 设置其相关参数学习率

**Comment by zhu143xin** [comment]
但是我上边说的都是封装好的，直接传入了参数，strategy传入的学习率，这边具体该怎么改能指导一下么，你说的那个需要在更深的源码里面修改吧

**Comment by zhu143xin** [comment]
> Hi，您可以参考 https://github.com/PaddlePaddle/PaddleHub/blob/release/v1.6/paddlehub/finetune/strategy.py#L389
> 通过判断param.name的方式 设置其相关参数学习率

但是我上边说的都是封装好的，直接传入了参数，strategy传入的学习率，这边具体该怎么改能指导一下么，你说的那个需要在更深的源码里面修改吧

**Comment by kinghuin** [comment]
如果不改源码的话您可以通过分层学习率的trick实现这个功能，

```python
from paddlehub.finetune.strategy import CombinedStrategy

class MyStrategy(CombinedStrategy): # 关于CombinedStrategy的详细说明请参考https://github.com/PaddlePaddle/PaddleHub/wiki/PaddleHub-API:-Strategy
    def __init__(self,
                 learning_rate=1e-4,
                 optimizer_name="adam",
                 factor=2.6,  # 请合理设置该参数, crf学习率将=learning_rate/factor
                ):

        scheduler = {
            "discriminative": {
                "factor": factor,
                "params_layer":  {"crfw":1,"fc_0.w_0":1,"fc_0.b_0":1,"trick_holder":2}
            }
        }
        regularization = {}
        clip = {}
        super(MyStrategy, self).__init__(
            optimizer_name=optimizer_name,
            learning_rate=learning_rate,
            scheduler=scheduler,
            regularization=regularization,
            clip=clip)
```

使用以上策略代替hub.AdamWeightDecayStrategy

**Comment by zhu143xin** [comment]
> 如果不改源码的话您可以通过分层学习率的trick实现这个功能，
> 
> ```python
> from paddlehub.finetune.strategy import CombinedStrategy
> 
> class MyStrategy(CombinedStrategy): # 关于CombinedStrategy的详细说明请参考https://github.com/PaddlePaddle/PaddleHub/wiki/PaddleHub-API:-Strategy
>     def __init__(self,
>                  learning_rate=1e-4,
>                  optimizer_name="adam",
>                  factor=2.6,  # 请合理设置该参数, crf学习率将=learning_rate/factor
>                 ):
> 
>         scheduler = {
>             "discriminative": {
>                 "factor": factor,
>                 "params_layer":  {"crfw":1,"fc_0.w_0":1,"fc_0.b_0":1,"trick_holder":2}
>             }
>         }
>         regularization = {}
>         clip = {}
>         super(MyStrategy, self).__init__(
>             optimizer_name=optimizer_name,
>             learning_rate=learning_rate,
>             scheduler=scheduler,
>             regularization=regularization,
>             clip=clip)
> ```
> 
> 使用以上策略代替hub.AdamWeightDecayStrategy

class MyStrategy(CombinedStrategy): # 关于CombinedStrategy的详细说明请参考https://github.com/PaddlePaddle/PaddleHub/wiki/PaddleHub-API:-Strategy
        def __init__(self,
                     learning_rate=3e-5,
                     optimizer_name="adam",
                     factor=0.01,  # 请合理设置该参数, crf学习率将=learning_rate/factor
                    ):

            scheduler = {
                "crf": {
                    "factor": factor,
                    "params_layer":  {"crfw":1,"fc_0.w_0":1,"fc_0.b_0":1,"trick_holder":2}
                }
            }
            regularization = {}
            clip = {}
            super(MyStrategy, self).__init__(
                optimizer_name=optimizer_name,
                learning_rate=learning_rate,
                warmup_proportion=args.warmup_proportion,
                weight_decay=args.weight_decay,
                scheduler=scheduler,
                regularization=regularization,
                clip=clip)

    # 选择优化策略
    # strategy = hub.AdamWeightDecayStrategy(
    #     warmup_proportion=args.warmup_proportion,
    #     weight_decay=args.weight_decay,
    #     learning_rate=args.learning_rate)

    # 配置运行设置
    config = hub.RunConfig(
            eval_interval=args.eval_step,
            save_ckpt_interval=args.model_save_step,
            use_data_parallel=args.use_data_parallel,
            use_cuda=True,
            num_epoch=args.num_epoch,
            batch_size=args.batch_size,
            checkpoint_dir=args.checkpoint_dir,
            strategy=MyStrategy)
这样子试了，报错TypeError: unbound method execute() must be called with MyStrategy instance as first argument (got Variable instance instead)

**Comment by kinghuin** [comment]
strategy=MyStrategy 应为strategy=MyStrategy()

**Comment by Steffy-zxf** [comment]
您好，此issue在近两周内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题。
